packages/open-truss/src/configuration/engine-v1/Frame.tsx

```diff
@@ -93,16 +92,6 @@ export function Frame(props: FrameContext): JSX.Element {
     })
     processedProps._DEBUG_ = debug
 
-    if (frames === undefined) {
-      if (data) {
-        return (
-          <DataProvider {...processedProps} data={data} component={Component} />
-        )
-      } else {
-        return <Component {...processedProps} />
```

Comment:
Not sure I understand how we can get rid of this part of this logic. Could you say more?

Reply:
Great question! With this block removed, the code will always return `<Component {...processedProps}>{subframes}</Component>`. When `subframes` is `undefined` because no `frames` have been specified in the annotation, it still renders just fine. i.e. `undefined` seems to be a valid component child.